src/pipes/db-lookup.pipe.ts

```diff
@@ -4,42 +4,31 @@ import {
   ValidationArguments,
 } from "class-validator";
 import { FindOneOptions } from "typeorm";
-import { DbLookupValidator } from '../validators';
+import { DbLookupValidator } from "../validators/db-lookup.validator";
+
+export interface DbLookupConfig {
+  type: any;
+  findOption?: ((validationArguments: ValidationArguments) => FindOneOptions) | "primary";
+  validateNumeric?: boolean;
+  ignoreEmpty?: boolean;
+}
 
 export function DbLookup(
-  type: any,
-  findOption?:
-    | ((validationArguments: ValidationArguments) => FindOneOptions)
-    | "primary"
-    | {
-        validateNumeric?: boolean;
-        ignoreEmpty?: boolean;
-        findOption?: ((validationArguments: ValidationArguments) => FindOneOptions) | "primary";
-      },
-  validationOptions?: ValidationOptions,
-  ignoreEmpty = true
+  config: DbLookupConfig,
+  validationOptions?: ValidationOptions
 ) {
   return function (object: Object, propertyName: string) {
     registerDecorator({
       name: "dbLookup",
       target: object.constructor,
       propertyName: propertyName,
       options: validationOptions,
-      constraints: [
-        typeof findOption === "object" && !Array.isArray(findOption)
-          ? {
-              ignoreEmpty: findOption.ignoreEmpty ?? ignoreEmpty,
-              validateNumeric: findOption.validateNumeric ?? false,
-              type,
-              findOption: findOption.findOption,
-            }
-          : {
-              ignoreEmpty,
-              validateNumeric: false,
-              type,
-              findOption,
-            },
-      ],
+      constraints: [{
+        ignoreEmpty: config.ignoreEmpty ?? true,
+        validateNumeric: config.validateNumeric ?? false,
+        type: config.type,
+        findOption: config.findOption
+      }],
```

Comment:
_If_ the approach noted above is used, this would need to change to:
```suggestion
      constraints: [{
        ignoreEmpty: config.ignoreEmpty ?? true,
        validateNumeric: config.validateNumeric ?? false,
        type,
        findOption,
      }],
```

Reply:
Pls see PR #170.